uint16 number
---
bool success
